### PR TITLE
Exposing Camera Euler Angles

### PR DESCRIPTION
**Pull request by devsherif**
For an app I am developing. I needed to position a 2D plane and rotate it initially to face the camera.
That's why I needed to access Camera's EulerAngles and I had to expose it using native Swift method.
My app is now working as required with the local changes I applied to the plugin. And thought about adding that change for others to use.

I hope my change is clean enough for being merged with your main repo.

Thanks,

### CODE REVIEW

File: ios/Classes/FlutterArkitView+Handlers.swift [hunk 1]
```diff
@@ -215,4 +215,17 @@ extension FlutterArkitView {
         }
         sceneView.scene.rootNode.removeAnimation(forKey: key)
     }
+
+    func onCameraEulerAngles(_ result:FlutterResult){
+        if let frame = sceneView.session.currentFrame {
+            let cameraEulerAngles = frame.camera.eulerAngles
+            var eulerAnglesArray: [Double] = [Double(cameraEulerAngles.x), Double(cameraEulerAngles.y), Double(cameraEulerAngles.z)]
+            let point = deserizlieVector3(eulerAnglesArray)
+            let res = serializeVector(point)
```
Comment (olexale) [inline ios/Classes/FlutterArkitView+Handlers.swift:224]: I propose to create an instance of `SCNVector3` and call `serializeVector()` on it.

File: lib/widget/arkit_scene_view.dart [hunk 2]
```diff
@@ -688,4 +688,11 @@ class ARKitController {
       ..addAll({paramName: params});
     return values;
   }
+
+  Future<Vector3> _getCameraEulerAngles() async{
```
Comment (olexale) [inline lib/widget/arkit_scene_view.dart:692]: Shouldn't this method be public?

File: lib/widget/arkit_scene_view.dart [hunk 3]
```diff
@@ -688,4 +688,11 @@ class ARKitController {
...
+    final result =  await _channel.invokeListMethod('cameraEulerAngles');
+    var vector3 = _vector3Converter.fromJson(result);
```
Comment (olexale) [inline lib/widget/arkit_scene_view.dart:694]: please make it `final` as well.

**Comment by olexale** [comment]
Hi @devsherif ,

Thank you for your contribution! 👍🏻

Please fix the comments above and I'll be glad to merge your changes to the repo!
Thanks in advance!

Kind regards,
Oleksandr

**Review by olexale — CHANGES_REQUESTED** [review]

**Comment by devsherif** [comment]
Hi @olexale ,

Thanks for your attention. 
Requested changes applied. Please check.

Regards,
Sherif

**Comment by olexale** [comment]
Hi @devsherif! 

I'll push a new version with your changes today. 
Thanks for your help!!

Kind regards,
Oleksandr

**Comment by devsherif** [comment]
Hi @olexale ,

Thanks a lot for your attention. 
And you can expect more changes soon ;)

Regards,
Sherif

**Comment by benMOOVENCY** [comment]
Hi @devsherif, 

have you a solution to expose the quaternion instead of the Euler angle to avoid the gimbal lock.

Cheers

**Pull request merged**